tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs

```diff
@@ -164,5 +164,48 @@ public async Task BuildsAndRunsNodeApp_WhenPruneDevDependenciesIsTrue_AndNodeMod
                     Assert.Contains("Say It Again", data);
                 });
         }
+
+        [Fact]
+        public async Task CopiesNodeModulesInSubDirectory_ToDestination_WhenNodeModulesAreCompressed()
+        {
+            // Arrange
+            // Use a separate volume for output due to rsync errors
+            var appOutputDirPath = Directory.CreateDirectory(Path.Combine(_tempRootDir, Guid.NewGuid().ToString("N")))
+                .FullName;
+            var nodeVersion = "10";
+            var appOutputDirVolume = DockerVolume.CreateMirror(appOutputDirPath);
+            var appOutputDir = appOutputDirVolume.ContainerDir;
+            var appName = "kudu-bug";
+            var volume = CreateAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var runAppScript = new ShellScriptBuilder()
+                .AddCommand($"oryx -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .AddDirectoryExistsCheck($"{appOutputDir}/another-directory/node_modules")
+                .ToString();
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand(
+                $"oryx build {appDir} -i /tmp/int -o {appOutputDir} --platform nodejs " +
+                $"--platform-version {nodeVersion} -p {NodePlatform.CompressNodeModulesPropertyKey}=tar-gz" +
+                $" -p {NodePlatform.PruneDevDependenciesPropertyKey}=true")
```

Comment:
Current app service does not use this, so i would say to to have multiple test cases here to test different scenarios here